nautobot/extras/tests/test_datasources.py

```diff
@@ -369,13 +377,13 @@ def test_pull_git_repository_and_refresh_data_with_valid_data(self, MockGitRepo)
                 MockGitRepo.return_value.checkout.return_value = self.COMMIT_HEXSHA
 
                 # Run the Git operation and refresh the object from the DB
-                pull_git_repository_and_refresh_data(self.repo.pk, self.mock_request, self.job_result.pk)
-                self.job_result.refresh_from_db()
-
+                job_model = GitRepositoryPullAndRefreshData().job_model
```

Comment:
Why do we need to instantiate the job-class in order to get its job-model? Seems like that should be a classproperty perhaps?

Same question below.

Reply:
Maybe. The problem we're having here is coming up with a pattern that allows it to be a cached property of some sort since it's making a database call, and we don't want it to make a database call every single time that attribute is accessed. Some kind of hybrid cached/classproperty would be ideal perhaps.

Reply:
Potentially for this line of code

```py
job_model = Job.objects.get(module_name=GitRepositoryPullAndRefreshData.__module__, job_class_name=GitRepositoryPullAndRefreshData.__name__)
```